### PR TITLE
FEAT: Implement AVLT_Insertion&Search and some extra visual features

### PR DESCRIPTION
**Pull request by JarrettChen217**
# FEAT: Implement AVLT_Insertion&Search and some extra visual features

This update enhances the Binary Search Tree (BST) by implementing the AVL Tree with the following modifications:

Screen Recording 2024-10-14 at 7 16 07 PM

## Changes Overview:
- [x] Added Pseudocode for AVL Tree insertion and search using a recursive approach.
- [x] Implemented AVL Tree Visualization
- [x] Added extra UI designs and animation for better visualization.
- [x] Fixed existing bugs in the Binary Search Tree implementation.

### UI & Animation Additions

image

#### Dynamic Information:
- **Height**: Displays the **current height** of the node.
- **Tid**: Indicates the `rotation pointer` of the current node.
- **Rectangle**: Highlight the **current unbalanced area**.

#### Static Information (Info Panel):
- **Function Name**: Shows recursion function call information.
- **Rotation Case**: Represents the current rotation case (showed during rotation replacing function name).
- **Function Balance**: Displays the balance factor of the current node.

#### Animation:
- **Rotation**: Added three new animation phases during rotation, focusing on the midpoint.

### Bug Fixes:

- **"Result Not Found" Error**: Resolved an issue where an error was thrown when an element could not be found during the search operation.

## Deployment Link

Vercel Deployment:

https://itp-team025-staticsound-bjl20879q-jarrettchen217s-projects.vercel.app/

### CODE REVIEW

**Comment by Sraccity** [comment]
@JarrettChen217  Just checking. This PR is ready to review correct?

> Alextroup-uom isn't active

**Comment by JarrettChen217** [comment]
@Sraccity Yes! We have resolved the conflicts, removed unnecessary signatures in-code and having our feature written in the description. Please let us know if you have any questions!

File: package.json [hunk 1]
```diff
@@ -15,7 +15,8 @@
     "start:win32": "SET NODE_OPTIONS=--openssl-legacy-provider && react-scripts start",
     "test": "react-scripts test",
     "test-uf": "npm test -- ./src/algorithms/controllers/unionFind.test.js",
-    "test-234t": "npm test -- ./src/algorithms/controllers/TTFTree.test.js"
+    "test-234t": "npm test -- ./src/algorithms/controllers/TTFTree.test.js",
+    "test_avl": "npm test -- ./src/algorithms/controllers/AVLTree.test.js"
```
Comment (Sraccity) [inline package.json:19]: Nitpick: Changing the format for standardisation

**Review by Sraccity — APPROVED** [review]
**Approved**

Good PR and thanks for your contributions

Kudos: Good exercises anyway.

Some changes will be done anyway 🫡

**Pull request merged**